docs/recipes/stencil.md

```diff
@@ -0,0 +1,87 @@
+# Usage with Stencil
+
+[Stencil](https://stenciljs.com/) web components work very well with XState.
+
+### `src/helpers/toggle-machine.ts`
+
+```js
+import { createMachine } from "@xstate/fsm"
+
+// this machine is completely decoupled from stencil
+export const toggleMachine = createMachine({
+  id: "toggle",
+  initial: "inactive",
+  states: {
+    inactive: {
+      on: { toggle: "active" }
+    },
+    active: {
+      on: { toggle: "inactive" }
+    }
+  }
+});
+```
+
+### `src/components/toggle/toggle.tsx`
+
+
+Add a `state` property to your component, decorated with `@State` so that it triggers a re-render when changed.
+
+On `componentWillLoad`, interpret the `toggleMachine` and listen for state transitions.
+
+After a transition has occured, the `state` property is set to the machine's new state, triggering a re-render.

```

Comment:
This make it sound like the state is not already set on the initial render? Would there be a way to get the initial state? Maybe sth like

```ts
export class Toggle {
  @State() state = toggleMachine.initialState;
}
```

(which would potentially even infer the correct `State` type?)

Reply:
I've made a codesandbox here btw to test this all works so far :-) 

https://codesandbox.io/s/stencil-xstate-7ll30

Great to see that stencil now works on codesandbox!

I think you may have to `interpret` first to get the initial state?

Reply:
Yeah Code Sandbox is pretty cool. Have a look at this one: https://codesandbox.io/s/stencil-xstate-wz0wq?file=/src/components/toggle/toggle.tsx

It correctly infers all the types, and no need to destructure `state` anymore. Only type info I added is for creating the machine.

```ts
export const toggleMachine = createMachine<{ toggle: "active" | "inactive" }>(/* ... */);
```

```ts
export class Toggle {
  @State() state = toggleMachine.initialState;

  private _service = interpret(toggleMachine);

  componentWillLoad() {
    this.service.subscribe(state => {
      this.state = state;
    });
    // ...
  }

  render() {
    return (
      // ...
        {this.state.value === "inactive" ? "Off" : "On"}
      // ...
    }
}
```

Reply:
Ok this was pretty wrong... i typed context instead of state and the initialState indeed doesn't work. But I think wrt types we can definitely still improve this recipe.

Reply:
Nice, have updated the PR to match this.